flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -267,6 +278,29 @@ public void addShipFiles(List<File> shipFiles) {
 		this.shipFiles.addAll(shipFiles);
 	}
 
+	/**
+	 * Adds the given files to the list of archives to ship.
+	 **
+	 * @param shipArchives archives to ship
+	 */
+	public void addShipArchives(List<File> shipArchives) {
+		checkArgument(isArchiveOnlyIncludedInShipArchiveFiles(shipArchives), "Non-archive files are included.");
+		this.shipArchives.addAll(shipArchives);
+	}
+
+	private static boolean isArchiveOnlyIncludedInShipArchiveFiles(List<File> shipFiles) {
+		return shipFiles.stream()
+			.filter(File::isFile)
+			.map(File::getName)
+			.map(String::toLowerCase)
+			.allMatch(name -> name.endsWith(".tar.gz") ||
+				name.endsWith(".tar") ||
+				name.endsWith(".tgz") ||
+				name.endsWith(".dst") ||
+				name.endsWith(".jar") ||
```

Comment:
Do we want to allow jars here ? This means that they will be un-jared at the other end.

Reply:
@kl0u I just checked all file types that yarn can support.
https://github.com/apache/hadoop/blob/8e4517f251e9f12917f2f457831664905015612f/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/util/FSDownload.java#L336-L343
If jars were disallowed, the reason should be documented. but, I am not sure why it should be removed.  😅

Reply:
Ok let's leave it like this for now :). Could you integrate the rest of my comments?

Reply:
I've done what you mentioned except adding a test to the YarnFileStageTest. I need a little more time in order to do that.  I'll push a commit about javadoc, style and etc first.